tests/test_pgen.rs

```diff
@@ -175,3 +175,102 @@ fn test_grammar2() {
     has(&follow, "Q", vec!["+",")"]);
     has(&follow, "F", vec!["*", "+", ")"]);
 }
+
+fn lritem(lhs: &str, rhs: Vec<Symbol>, d: usize) -> LR1Item {
+    let mut item = LR1Item::new(lhs.to_string(), Vec::new(), d);
+    for e in rhs.iter() {
+        item.rhs.push(e.clone());
+    }
+    item
+}
+
+fn vec_to_set(vec: Vec<&str>) -> HashSet<String> {
```

Comment:
I'm not sure this name is great. It suggests it's a generic vec_to_set which would presumably have a type signature vec_to_set<T>(vec: Vec<T>) -> HashSet<T>. Maybe call it mk_str_set or something like that?